episodes/12-handling-spatial-projection-and-crs.Rmd

```diff
@@ -0,0 +1,137 @@
+  filter(leisure %in% c("playground", "picnic_table"))
+```
+
+:::::::::::::::::::::::::::::::::::::: questions 
+
+- What do I do when vector data don’t line up?
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: objectives
+
+- Plot vector objects with different CRSs in the same plot.
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+## Working with spatial data from different sources
+
+```{r}
+municipal_boundary_NL <- st_read("data/nl-gemeenten.shp")
+```
+
+```{r}
+ggplot() +
+  geom_sf(data = municipal_boundary_NL) +
+  labs(title = "Map of Contiguous NL Municipal Boundaries") +
+  coord_sf(datum = st_crs(28992))
+```
+
+We can add a country boundary layer to make it look nicer. If we specify a thicker line width using size = 2 for the country boundary layer, it will make our map pop!
+
+```{r}
+country_boundary_NL <- st_read("data/nl-boundary.shp")
+```
+
+```{r}
+ggplot() +
+  geom_sf(data = country_boundary_NL, color = "gray18", linewidth = 2) +
+  geom_sf(data = municipal_boundary_NL, color = "gray40") +
+  labs(title = "Map of Contiguous NL Municipal Boundaries") +
+  coord_sf(datum = st_crs(28992))
+```
+
+```{r}
+# st_crs(point_Delft)
+```
+
+```{r}
+st_crs(municipal_boundary_NL)$epsg
+```
+
+```{r}
+st_crs(country_boundary_NL)$epsg
+```
+
+```{r}
+boundary_Delft <- st_read("data/delft-boundary.shp")
+st_crs(boundary_Delft)$epsg
+boundary_Delft <- st_transform(boundary_Delft, 28992)
+```
+
+```{r}
+ggplot() +
+  geom_sf(data = country_boundary_NL, linewidth = 2, color = "gray18") +
+  geom_sf(data = municipal_boundary_NL, color = "gray40") +
+  geom_sf(data = boundary_Delft, color = "purple", fill = "purple") +
+  labs(title = "Map of Contiguous NL Municipal Boundaries") +
+  coord_sf(datum = st_crs(28992))
+```
+
+::: challenge
+
+# Plot multiple layers of spatial data
+<!-- 5 minutes -->
+
+Create a map of South Holland as follows:
+
+1. Import and plot `nl-gemeenten.shp`. Adjust line width as necessary.
+2. Layer the boundary of Delft onto the plot.
+3. Add a title.
+4. Add a legend that shows both the province boundaries (as a line) and the boundary of Delft (as a filled polygon).
+
+::: solution
+
+```{r}
+boundary_ZH <- municipal_boundary_NL %>% 
+  filter(ligtInPr_1 == "Zuid-Holland")
+```
+
+```{r}
+ggplot() +
+    geom_sf(data = boundary_ZH, aes(color ="color"), show.legend = "line") +
+    scale_color_manual(name = "", labels = "Municipal Boundaries in South Holland", values = c("color" = "gray18")) +
+    geom_sf(data = boundary_Delft, aes(shape = "shape"), color = "purple", fill = "purple") +
+    scale_shape_manual(name = "", labels = "Municipality of Delft", values = c("shape" = 19)) +
+    labs(title = "Delft location") +
+    theme(legend.background = element_rect(color = NA)) +
+    coord_sf(datum = st_crs(28992))
+```
+
+:::
+
+:::
+
+
+
+## Export a shapefile
+
+To save a file, use the `st_write()` function from the `sf` package.
```

Comment:
```suggestion
To save a file, use the `st_write()` function from the `sf` package. Although `sf` guesses the driver needed for a specified output file name from its extension, this can be made explicitly via the `driver` argument. In our case `driver = "ESRI Shapefile"` ensures that the the output is correctly saved as a `.shp` file.
```